{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Personal tweets sentiment analysis\n",
    "\n",
    "## This paper presents user interaction with the developed model\n",
    "\n",
    "interaction interface:\n",
    "1: You will need to enter screenname in the variable \"user_screen_name\"<br>\n",
    "2: Enter the required number of current tweets to view the variable \"tweets_count\"<br>\n",
    "3: Run all jupyter cells<br>\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "user_screen_name = \"realDonaldTrump\"\n",
    "tweets_count = 5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/vova/.local/lib/python3.6/site-packages/ipykernel_launcher.py:111: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "/home/vova/.local/lib/python3.6/site-packages/ipykernel_launcher.py:113: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<tf.Variable 'embedding:0' shape=(109, 200) dtype=float64_ref>\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/vova/.local/lib/python3.6/site-packages/ipykernel_launcher.py:140: DeprecationWarning: Call to deprecated `layer1_size` (Attribute will be removed in 4.0.0, use self.trainables.layer1_size instead).\n",
      "/home/vova/.local/lib/python3.6/site-packages/ipykernel_launcher.py:156: DeprecationWarning: Call to deprecated `__getitem__` (Method will be removed in 4.0.0, use self.wv.__getitem__() instead).\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exception in thread Reloader:\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 864, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/application.py\", line 361, in _reload_forever\n",
      "    reload_multiplexer(multiplexer, path_to_run)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/application.py\", line 335, in reload_multiplexer\n",
      "    multiplexer.Reload()\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/plugin_event_multiplexer.py\", line 195, in Reload\n",
      "    accumulator.Reload()\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/plugin_event_accumulator.py\", line 185, in Reload\n",
      "    for event in self._generator.Load():\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/directory_watcher.py\", line 88, in Load\n",
      "    for event in self._LoadInternal():\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/directory_watcher.py\", line 112, in _LoadInternal\n",
      "    for event in self._loader.Load():\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/event_file_loader.py\", line 53, in Load\n",
      "    self._reader.GetNext(status)\n",
      "TypeError: GetNext() takes 1 positional argument but 2 were given\n",
      "\n",
      "TensorBoard 1.5.1 at http://vova-Aspire-A515-51G:8080 (Press CTRL+C to quit)\n",
      "\u001b[33mW0102 17:01:01.016032 Thread-1 application.py:273] path /[[_dataImageSrc]] not found, sending 404\n",
      "\u001b[0m\u001b[31mE0102 17:01:01.858206 Thread-1 _internal.py:88] Error on request:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/werkzeug/serving.py\", line 270, in run_wsgi\n",
      "    execute(self.server.app)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/werkzeug/serving.py\", line 258, in execute\n",
      "    application_iter = app(environ, start_response)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/application.py\", line 271, in __call__\n",
      "    return self.data_applications[clean_path](environ, start_response)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/werkzeug/wrappers.py\", line 308, in application\n",
      "    resp = f(*args[:-2] + (request,))\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/plugins/core/core_plugin.py\", line 136, in _serve_runs\n",
      "    for run_name in run_names\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/plugins/core/core_plugin.py\", line 136, in <dictcomp>\n",
      "    for run_name in run_names\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/plugins/core/core_plugin.py\", line 124, in get_first_event_timestamp\n",
      "    return self._multiplexer.FirstEventTimestamp(run_name)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/plugin_event_multiplexer.py\", line 259, in FirstEventTimestamp\n",
      "    return accumulator.FirstEventTimestamp()\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/plugin_event_accumulator.py\", line 235, in FirstEventTimestamp\n",
      "    event = next(self._generator.Load())\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/directory_watcher.py\", line 88, in Load\n",
      "    for event in self._LoadInternal():\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/directory_watcher.py\", line 112, in _LoadInternal\n",
      "    for event in self._loader.Load():\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/event_file_loader.py\", line 53, in Load\n",
      "    self._reader.GetNext(status)\n",
      "TypeError: GetNext() takes 1 positional argument but 2 were given\n",
      "\u001b[0m\u001b[31mE0102 17:01:01.877412 Thread-3 _internal.py:88] Error on request:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/werkzeug/serving.py\", line 270, in run_wsgi\n",
      "    execute(self.server.app)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/werkzeug/serving.py\", line 258, in execute\n",
      "    application_iter = app(environ, start_response)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/application.py\", line 271, in __call__\n",
      "    return self.data_applications[clean_path](environ, start_response)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/werkzeug/wrappers.py\", line 308, in application\n",
      "    resp = f(*args[:-2] + (request,))\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/plugins/core/core_plugin.py\", line 136, in _serve_runs\n",
      "    for run_name in run_names\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/plugins/core/core_plugin.py\", line 136, in <dictcomp>\n",
      "    for run_name in run_names\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/plugins/core/core_plugin.py\", line 124, in get_first_event_timestamp\n",
      "    return self._multiplexer.FirstEventTimestamp(run_name)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/plugin_event_multiplexer.py\", line 259, in FirstEventTimestamp\n",
      "    return accumulator.FirstEventTimestamp()\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/plugin_event_accumulator.py\", line 235, in FirstEventTimestamp\n",
      "    event = next(self._generator.Load())\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/directory_watcher.py\", line 88, in Load\n",
      "    for event in self._LoadInternal():\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/directory_watcher.py\", line 112, in _LoadInternal\n",
      "    for event in self._loader.Load():\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/event_file_loader.py\", line 53, in Load\n",
      "    self._reader.GetNext(status)\n",
      "TypeError: GetNext() takes 1 positional argument but 2 were given\n",
      "\u001b[0m\u001b[31mE0102 17:01:01.881289 Thread-3 _internal.py:88] Error on request:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/werkzeug/serving.py\", line 270, in run_wsgi\n",
      "    execute(self.server.app)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/werkzeug/serving.py\", line 258, in execute\n",
      "    application_iter = app(environ, start_response)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/application.py\", line 271, in __call__\n",
      "    return self.data_applications[clean_path](environ, start_response)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/werkzeug/wrappers.py\", line 308, in application\n",
      "    resp = f(*args[:-2] + (request,))\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/plugins/core/core_plugin.py\", line 136, in _serve_runs\n",
      "    for run_name in run_names\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/plugins/core/core_plugin.py\", line 136, in <dictcomp>\n",
      "    for run_name in run_names\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/plugins/core/core_plugin.py\", line 124, in get_first_event_timestamp\n",
      "    return self._multiplexer.FirstEventTimestamp(run_name)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/plugin_event_multiplexer.py\", line 259, in FirstEventTimestamp\n",
      "    return accumulator.FirstEventTimestamp()\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/plugin_event_accumulator.py\", line 235, in FirstEventTimestamp\n",
      "    event = next(self._generator.Load())\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/directory_watcher.py\", line 88, in Load\n",
      "    for event in self._LoadInternal():\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/directory_watcher.py\", line 112, in _LoadInternal\n",
      "    for event in self._loader.Load():\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/event_file_loader.py\", line 53, in Load\n",
      "    self._reader.GetNext(status)\n",
      "TypeError: GetNext() takes 1 positional argument but 2 were given\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[31mE0102 17:01:01.990921 Thread-3 _internal.py:88] Error on request:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/werkzeug/serving.py\", line 270, in run_wsgi\n",
      "    execute(self.server.app)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/werkzeug/serving.py\", line 258, in execute\n",
      "    application_iter = app(environ, start_response)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/application.py\", line 271, in __call__\n",
      "    return self.data_applications[clean_path](environ, start_response)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/werkzeug/wrappers.py\", line 308, in application\n",
      "    resp = f(*args[:-2] + (request,))\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/plugins/core/core_plugin.py\", line 136, in _serve_runs\n",
      "    for run_name in run_names\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/plugins/core/core_plugin.py\", line 136, in <dictcomp>\n",
      "    for run_name in run_names\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/plugins/core/core_plugin.py\", line 124, in get_first_event_timestamp\n",
      "    return self._multiplexer.FirstEventTimestamp(run_name)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/plugin_event_multiplexer.py\", line 259, in FirstEventTimestamp\n",
      "    return accumulator.FirstEventTimestamp()\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/plugin_event_accumulator.py\", line 235, in FirstEventTimestamp\n",
      "    event = next(self._generator.Load())\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/directory_watcher.py\", line 88, in Load\n",
      "    for event in self._LoadInternal():\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/directory_watcher.py\", line 112, in _LoadInternal\n",
      "    for event in self._loader.Load():\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/event_file_loader.py\", line 53, in Load\n",
      "    self._reader.GetNext(status)\n",
      "TypeError: GetNext() takes 1 positional argument but 2 were given\n",
      "\u001b[0m\u001b[31mE0102 17:01:02.392305 Thread-3 _internal.py:88] Error on request:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/werkzeug/serving.py\", line 270, in run_wsgi\n",
      "    execute(self.server.app)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/werkzeug/serving.py\", line 258, in execute\n",
      "    application_iter = app(environ, start_response)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/application.py\", line 271, in __call__\n",
      "    return self.data_applications[clean_path](environ, start_response)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/werkzeug/wrappers.py\", line 308, in application\n",
      "    resp = f(*args[:-2] + (request,))\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/plugins/core/core_plugin.py\", line 136, in _serve_runs\n",
      "    for run_name in run_names\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/plugins/core/core_plugin.py\", line 136, in <dictcomp>\n",
      "    for run_name in run_names\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/plugins/core/core_plugin.py\", line 124, in get_first_event_timestamp\n",
      "    return self._multiplexer.FirstEventTimestamp(run_name)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/plugin_event_multiplexer.py\", line 259, in FirstEventTimestamp\n",
      "    return accumulator.FirstEventTimestamp()\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/plugin_event_accumulator.py\", line 235, in FirstEventTimestamp\n",
      "    event = next(self._generator.Load())\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/directory_watcher.py\", line 88, in Load\n",
      "    for event in self._LoadInternal():\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/directory_watcher.py\", line 112, in _LoadInternal\n",
      "    for event in self._loader.Load():\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/event_file_loader.py\", line 53, in Load\n",
      "    self._reader.GetNext(status)\n",
      "TypeError: GetNext() takes 1 positional argument but 2 were given\n",
      "\u001b[0m\u001b[31mE0102 17:01:02.395082 Thread-3 _internal.py:88] Error on request:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/werkzeug/serving.py\", line 270, in run_wsgi\n",
      "    execute(self.server.app)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/werkzeug/serving.py\", line 258, in execute\n",
      "    application_iter = app(environ, start_response)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/application.py\", line 271, in __call__\n",
      "    return self.data_applications[clean_path](environ, start_response)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/werkzeug/wrappers.py\", line 308, in application\n",
      "    resp = f(*args[:-2] + (request,))\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/plugins/core/core_plugin.py\", line 136, in _serve_runs\n",
      "    for run_name in run_names\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/plugins/core/core_plugin.py\", line 136, in <dictcomp>\n",
      "    for run_name in run_names\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/plugins/core/core_plugin.py\", line 124, in get_first_event_timestamp\n",
      "    return self._multiplexer.FirstEventTimestamp(run_name)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/plugin_event_multiplexer.py\", line 259, in FirstEventTimestamp\n",
      "    return accumulator.FirstEventTimestamp()\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/plugin_event_accumulator.py\", line 235, in FirstEventTimestamp\n",
      "    event = next(self._generator.Load())\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/directory_watcher.py\", line 88, in Load\n",
      "    for event in self._LoadInternal():\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/directory_watcher.py\", line 112, in _LoadInternal\n",
      "    for event in self._loader.Load():\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/event_file_loader.py\", line 53, in Load\n",
      "    self._reader.GetNext(status)\n",
      "TypeError: GetNext() takes 1 positional argument but 2 were given\n",
      "\u001b[0m\u001b[31mE0102 17:01:02.473577 Thread-4 _internal.py:88] Error on request:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/werkzeug/serving.py\", line 270, in run_wsgi\n",
      "    execute(self.server.app)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/werkzeug/serving.py\", line 258, in execute\n",
      "    application_iter = app(environ, start_response)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/application.py\", line 271, in __call__\n",
      "    return self.data_applications[clean_path](environ, start_response)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/werkzeug/wrappers.py\", line 308, in application\n",
      "    resp = f(*args[:-2] + (request,))\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/plugins/core/core_plugin.py\", line 136, in _serve_runs\n",
      "    for run_name in run_names\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/plugins/core/core_plugin.py\", line 136, in <dictcomp>\n",
      "    for run_name in run_names\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/plugins/core/core_plugin.py\", line 124, in get_first_event_timestamp\n",
      "    return self._multiplexer.FirstEventTimestamp(run_name)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/plugin_event_multiplexer.py\", line 259, in FirstEventTimestamp\n",
      "    return accumulator.FirstEventTimestamp()\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/plugin_event_accumulator.py\", line 235, in FirstEventTimestamp\n",
      "    event = next(self._generator.Load())\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/directory_watcher.py\", line 88, in Load\n",
      "    for event in self._LoadInternal():\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/directory_watcher.py\", line 112, in _LoadInternal\n",
      "    for event in self._loader.Load():\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/event_file_loader.py\", line 53, in Load\n",
      "    self._reader.GetNext(status)\n",
      "TypeError: GetNext() takes 1 positional argument but 2 were given\n",
      "\u001b[0m\u001b[31mE0102 17:01:02.477700 Thread-4 _internal.py:88] Error on request:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/werkzeug/serving.py\", line 270, in run_wsgi\n",
      "    execute(self.server.app)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/werkzeug/serving.py\", line 258, in execute\n",
      "    application_iter = app(environ, start_response)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/application.py\", line 271, in __call__\n",
      "    return self.data_applications[clean_path](environ, start_response)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/werkzeug/wrappers.py\", line 308, in application\n",
      "    resp = f(*args[:-2] + (request,))\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/plugins/core/core_plugin.py\", line 136, in _serve_runs\n",
      "    for run_name in run_names\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/plugins/core/core_plugin.py\", line 136, in <dictcomp>\n",
      "    for run_name in run_names\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/plugins/core/core_plugin.py\", line 124, in get_first_event_timestamp\n",
      "    return self._multiplexer.FirstEventTimestamp(run_name)\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/plugin_event_multiplexer.py\", line 259, in FirstEventTimestamp\n",
      "    return accumulator.FirstEventTimestamp()\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/plugin_event_accumulator.py\", line 235, in FirstEventTimestamp\n",
      "    event = next(self._generator.Load())\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/directory_watcher.py\", line 88, in Load\n",
      "    for event in self._LoadInternal():\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/directory_watcher.py\", line 112, in _LoadInternal\n",
      "    for event in self._loader.Load():\n",
      "  File \"/home/vova/.local/lib/python3.6/site-packages/tensorboard/backend/event_processing/event_file_loader.py\", line 53, in Load\n",
      "    self._reader.GetNext(status)\n",
      "TypeError: GetNext() takes 1 positional argument but 2 were given\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "from apiclient import discovery\n",
    "from httplib2 import Http\n",
    "import oauth2client\n",
    "from oauth2client import file, client, tools\n",
    "import io\n",
    "from googleapiclient.http import MediaIoBaseDownload\n",
    "import tweepy\n",
    "import csv\n",
    "import pandas as pd\n",
    "from bs4 import BeautifulSoup\n",
    "from nltk.tokenize import WordPunctTokenizer\n",
    "from nltk.stem.wordnet import WordNetLemmatizer\n",
    "from nltk.stem.lancaster import LancasterStemmer\n",
    "from gensim.models import Word2Vec\n",
    "import multiprocessing\n",
    "from nltk.corpus import stopwords\n",
    "import re\n",
    "import pickle\n",
    "import numpy as np\n",
    "from sklearn.model_selection import train_test_split\n",
    "import pandas as pd\n",
    "import tensorflow as tf\n",
    "from tensorflow.contrib.tensorboard.plugins import projector\n",
    "import numpy as np\n",
    "import gensim\n",
    "import random\n",
    "import os\n",
    "import warnings\n",
    "warnings.filterwarnings(action='ignore', category=UserWarning, module='gensim')\n",
    "pd.set_option('display.max_columns', 7)\n",
    "\n",
    "obj = lambda: None\n",
    "lmao = {\"auth_host_name\":'localhost', 'noauth_local_webserver':'store_true', 'auth_host_port':[8080, 8090], 'logging_level':'ERROR'}\n",
    "for k, v in lmao.items():\n",
    "    setattr(obj, k, v)\n",
    "    \n",
    "SCOPES = 'https://www.googleapis.com/auth/drive.readonly'\n",
    "store = file.Storage('token.json')\n",
    "creds = store.get()\n",
    "\n",
    "if not creds or creds.invalid:\n",
    "    flow = client.flow_from_clientsecrets('client_id.json', SCOPES)\n",
    "    creds = tools.run_flow(flow, store, obj)\n",
    "\n",
    "DRIVE = discovery.build('drive', 'v3', http=creds.authorize(Http()))\n",
    "\n",
    "file_id = '1gN9u4zFWfwR5n-LmBwrcwmNGIUKj4Y0F'\n",
    "request = DRIVE.files().get_media(fileId=file_id)\n",
    "\n",
    "fh = io.FileIO('lemmatization_nolim_all.sav', mode='w')\n",
    "downloader = MediaIoBaseDownload(fh, request)\n",
    "done = False\n",
    "while done is False:\n",
    "    status, done = downloader.next_chunk()\n",
    "    print(\"Download %d%%.\" % int(status.progress() * 100))\n",
    "\n",
    "predict_model = 'lemmatization_nolim_all.sav'\n",
    "parent_patch = os.getcwd()\n",
    "path = \"test\"\n",
    "stopWords = set(stopwords.words('english'))\n",
    "\n",
    "consumer_key = 'NgbszsMy18esxzBRpnS6YJSg5'\n",
    "consumer_secret = 'fUlGwElm7B7Q5UUl99TdnMewBA3xW9Cw5xmzBAq1xU9j5O6wUa'\n",
    "access_key = '3847979172-1TNy6qbn1DvF2lHuUMpM86hAyRSxN8Uc9WpZzET'  # access_token\n",
    "access_secret = 'ZCooGbFqAqxCyFtZGqMPczAhD6IkZW1TfT1hocKVPm8pV'\n",
    "\n",
    "tok = WordPunctTokenizer()\n",
    "\n",
    "pat1 = r'@[A-Za-z0-9_]+'\n",
    "pat2 = r'https?://[^ ]+'\n",
    "combined_pat = r'|'.join((pat1, pat2))\n",
    "www_pat = r'www.[^ ]+'\n",
    "negations_dic = {\"isn't\": \"is not\", \"aren't\": \"are not\", \"wasn't\": \"was not\", \"weren't\": \"were not\",\n",
    "                 \"haven't\": \"have not\", \"hasn't\": \"has not\", \"hadn't\": \"had not\", \"won't\": \"will not\",\n",
    "                 \"wouldn't\": \"would not\", \"don't\": \"do not\", \"doesn't\": \"does not\", \"didn't\": \"did not\",\n",
    "                 \"can't\": \"can not\", \"couldn't\": \"could not\", \"shouldn't\": \"should not\", \"mightn't\": \"might not\",\n",
    "                 \"mustn't\": \"must not\"}\n",
    "neg_pattern = re.compile(r'\\b(' + '|'.join(negations_dic.keys()) + r')\\b')\n",
    "\n",
    "\n",
    "def tweet_cleaner_updated(text, tweet_len=100):\n",
    "    soup = BeautifulSoup(text, 'lxml')\n",
    "    souped = soup.get_text()\n",
    "    try:\n",
    "        bom_removed = souped.decode(\"utf-8-sig\").replace(u\"\\ufffd\", \"?\")\n",
    "    except:\n",
    "        bom_removed = souped\n",
    "    stripped = re.sub(combined_pat, '', bom_removed)\n",
    "    stripped = re.sub(www_pat, '', stripped)\n",
    "    lower_case = stripped.lower()\n",
    "    neg_handled = neg_pattern.sub(lambda x: negations_dic[x.group()], lower_case)\n",
    "    letters_only = re.sub(\"[^a-zA-Z]\", \" \", neg_handled)\n",
    "\n",
    "    lema = WordNetLemmatizer()\n",
    "    lancaster_stemmer = LancasterStemmer()\n",
    "    words = list()\n",
    "    for word in tok.tokenize(letters_only):\n",
    "        if len(word) > 1 and word not in stopWords:\n",
    "            #print('raw', word)\n",
    "            lema_word = lema.lemmatize(word)\n",
    "            #lema_word = lancaster_stemmer.stem(word)\n",
    "            if len(lema_word) == 1:\n",
    "                lema_word = word\n",
    "            #print('lem', lema_word)\n",
    "            words.append(lema_word)\n",
    "    if len(words) <= tweet_len:\n",
    "        return words, (\" \".join(words)).strip()\n",
    "\n",
    "\n",
    "def get_api_clean_tweets_df(screen_name, tweet_num=5, predict_model=''):\n",
    "    auth = tweepy.OAuthHandler(consumer_key, consumer_secret)\n",
    "    auth.set_access_token(access_key, access_secret)\n",
    "    api = tweepy.API(auth)\n",
    "\n",
    "    columns = ['Screen_Name', 'Time_Stamp', 'raw_tweet', 'text', 'tokens']\n",
    "    tweet_df = pd.DataFrame(columns=columns)\n",
    "    tweet_tokeenized = list()\n",
    "    tweet_tokens = list()\n",
    "    positive_tokens = list()\n",
    "    tokens_positiveti_dict = dict()\n",
    "\n",
    "    index = 0\n",
    "    for status in tweepy.Cursor(api.user_timeline, screen_name=screen_name, tweet_mode=\"extended\").items():\n",
    "        clean_tweet = tweet_cleaner_updated(status.full_text)\n",
    "        tweet_tokens += clean_tweet[0]\n",
    "        tweet_tokeenized.append(clean_tweet[0])\n",
    "        tweet_df.loc[index] = [status.user.screen_name, status.created_at,\n",
    "                               status.full_text, clean_tweet[1], clean_tweet[0]]\n",
    "        index += 1\n",
    "        if index == tweet_num:\n",
    "            break\n",
    "            \n",
    "    tweet_df['target'] = None\n",
    "\n",
    "    loaded_model = pickle.load(open(predict_model, 'rb'))\n",
    "\n",
    "    for index in range(len(tweet_df)):\n",
    "        if tweet_df['text'][index] is np.nan:\n",
    "            pass\n",
    "        else:\n",
    "            if int(loaded_model.predict(list([tweet_df['text'][index]]))) == 1:\n",
    "                tweet_df['target'][index] = 'Postive'\n",
    "            else:\n",
    "                tweet_df['target'][index] = 'Negative'\n",
    "\n",
    "\n",
    "    positive_tweet_df = tweet_df.loc[tweet_df['target'] == 'Postive']\n",
    "    for row in  positive_tweet_df.index:  # range(len(positive_tweet_df)):\n",
    "        positive_tokens += positive_tweet_df['tokens'][row]\n",
    "\n",
    "    tweet_tokens_set = set(tweet_tokens)\n",
    "    for token in tweet_tokens_set:\n",
    "        token_all_counter = tweet_tokens.count(token)\n",
    "        token_pos_counter = positive_tokens.count(token)\n",
    "        tokens_positiveti_dict[token] = int(token_pos_counter/token_all_counter*100)\n",
    "\n",
    "    tweet_df.drop(['Screen_Name', 'Time_Stamp'], axis=1, inplace=True)\n",
    "    return tokens_positiveti_dict ,tweet_tokens, tweet_tokeenized, tweet_df\n",
    "\n",
    "\n",
    "pos_tokens_dict, donalds_tokens_list, donalds_tokenized_tweets, donalds_df =\\\n",
    "    get_api_clean_tweets_df(user_screen_name, tweets_count, predict_model)\n",
    "\n",
    "cores = multiprocessing.cpu_count()\n",
    "\n",
    "user_model = Word2Vec(donalds_tokenized_tweets, min_count=1, size=200, workers=cores,)\n",
    "user_model.save(\"user_model\")\n",
    "model = gensim.models.keyedvectors.KeyedVectors.load(\"user_model\")\n",
    "\n",
    "max_size = len(model.wv.vocab)-1\n",
    "w2v = np.zeros((max_size, model.layer1_size))\n",
    "\n",
    "with open(\"test/metadata.tsv\", 'w+') as file_metadata:\n",
    "    meta_word = ('word' + '\\t' + 'Sentiment')\n",
    "    file_metadata.write(meta_word + '\\n')\n",
    "\n",
    "with open('tensors.tsv', 'w+') as tensors:\n",
    "    with open(\"test/metadata.tsv\", 'a') as file_metadata:\n",
    "        for i, word in enumerate(model.wv.index2word[:max_size]):\n",
    "            w2v[i] = model.wv[word]\n",
    "            if pos_tokens_dict[word] < 50:\n",
    "                meta_word = word +'('+str(pos_tokens_dict[word])+')'+'\\t'+str(0)\n",
    "                file_metadata.write(meta_word + '\\n')\n",
    "            else:\n",
    "                meta_word = word +'('+str(pos_tokens_dict[word])+')'+'\\t'+str(100)\n",
    "                file_metadata.write(meta_word + '\\n')\n",
    "            vector_row = '\\t'.join(map(str, model[word]))\n",
    "            tensors.write(vector_row + '\\n')\n",
    "\n",
    "\n",
    "sess = tf.InteractiveSession()\n",
    "\n",
    "with tf.device(\"/cpu:0\"):\n",
    "    embedding = tf.Variable(w2v, trainable=False, name='embedding')\n",
    "    print(embedding)\n",
    "\n",
    "tf.global_variables_initializer().run()\n",
    "\n",
    "\n",
    "saver = tf.train.Saver()\n",
    "\n",
    "writer = tf.summary.FileWriter(path, sess.graph)\n",
    "\n",
    "config = projector.ProjectorConfig()\n",
    "embed = config.embeddings.add()\n",
    "embed.tensor_name = 'embedding'\n",
    "embed.metadata_path = 'metadata.tsv'\n",
    "\n",
    "projector.visualize_embeddings(writer, config)\n",
    "\n",
    "saver.save(sess, path+'/model.ckpt', global_step=max_size)\n",
    "\n",
    "! tensorboard --logdir=\"test\" --port=8080\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
